### PR TITLE
dafny: pin z3 and Boogie version

### PR DESCRIPTION
**Pull request by layus**
###### Motivation for this change

Dafny was broken by mismatched updates to z3 and Boogie

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by taktoa — APPROVED** [review]

**Comment by taktoa** [comment]
Sorry, no idea how I didn't notice this.

**Comment by layus** [comment]
thanks :-)

**Review by samuela — APPROVED** [review]
LGTM nixpkgs-review passes

File: pkgs/top-level/dotnet-packages.nix [hunk 1]
```diff
@@ -326,14 +326,14 @@ let self = dotnetPackages // overrides; dotnetPackages = with self; {
 
   Boogie = buildDotnetPackage rec {
     baseName = "Boogie";
-    version = "2019-06-20";
-    name = "${baseName}-unstable-${version}";
+    version = "2.4.1";
+    name = "${baseName}-${version}";
```
Comment (AndersonTorres) [inline pkgs/top-level/dotnet-packages.nix:330]: `pname`, maybe?

Comment (layus) [reply pkgs/top-level/dotnet-packages.nix]: well, baseName amounts to pname in dotnetPackages, but `name = "${baseName}-${version}";` is completely redundant with buildSotnetPackage indeed.

**Comment by layus** [comment]
I rebased and refreshed the commits. Should be good to go :-)

**Pull request merged**